doc/overview.md

```diff
@@ -0,0 +1,76 @@
+
+ * Standards-based: the analyzer ships with a set of scanners that are able to parse and analyze standards-based web sites.
+
+ * Loading agnostic: the user specifies both how to load files, and how URLs map into paths for the loader. This enables use cases like loading files over a network, as well as cases where the filesystem does not map 1:1 onto the url space as the source code sees it.
+
+## Design
+
+### Phases
+
+From a high level, the analyzer's work is broken up into phases.
+
+    Loading ➡ Parsing ➡ Scanning ➡ Resolving
+
+*Loading* takes a URL and returns a string representing the file's contents.
+
+*Parsing* takes that string and returns an abstract syntax tree that represents the raw syntax of the code.
+
+*Scanning* takes that AST and returns higher-level, more semantically meaningful, file-local features. While scanning, we may encounter inline documents and imports, which will cause them to be loaded, parsed, and scanned as well. A document isn't finished being scanned until all of its dependencies are also scanned.
+
+*Resolving* takes a scanned feature and all resolved features found up until that point and returns a resolved feature. Resolving can use information defined earlier in the file, or in files imported earlier in the file. It's the only point in the analysis process that can use cross-file information.
+
+
+### Example
+
+Let's walk through those steps using this file as an example:
+
+```html
+  <!-- progress-bar.html -->
+  <link rel="import" href="../polymer/polymer.html">
+
+  <script>
+    class ProgressBar extends Polymer.Element {
+      doSomethingCool() {
+        /** ... */
+      }
+    }
+  </script>
+```
+
+Loading is just the process of turning `'progress-bar.html'` into the source code above as a string.
+
+Parsing will convert that into an HTML document with two elements and some whitespace text nodes.
+
+The scanning process will convert that into a ScannedImport and a ScannedInlineDocument. The analyzer will kick off loading, parsing, and scanning of `polymer.html` as well as parsing and scanning the inline javascript document.
+
+The inline JS is parsed into an AST containing a class declaration statement. The scanners will extract a ScannedClass. This ScannedClass will have a reference to the super class `Polymer.Element`, but the reference will just note the name `Polymer.Element`, it won't have any more info than that because scanning is file-local. The ScannedClass will also note that `ProgressBar` has a method called `doSomethingCool` that takes no parameters.
+
+At resolution time the ScannedClass is able to follow the reference, assuming that `polymer.html` defines a class named `Polymer.Element`. `ScannedClass.resolve()` models javascript inheritance, starting with the methods and properties of `Polymer.Element` and then overriding and augmenting them with those on `ProgressBar` before returning a resolved `Class` feature.
```

Comment:
Above I see:
-  "Loading is just..."
- "Parsing will..."
- "The scanning process will..."
- "At resolution time..."

Give these all the same structure (and even italicize the keyword) so that I can anchor back to the list of steps you just gave me.